run_experiments.py

```diff
@@ -463,7 +481,8 @@ def main():
         if not log_project(project, source_dir, args.jobs):
             continue
         check_project(project, source_dir, config, args.jobs)
-        post_process_project(project, source_dir, config, printer)
+        post_process_project(project, source_dir, config, printer,
```

Comment:
I would prefer to make post_process return whether there were any errors or asserts and exit here based on `fail_on_assert`.  This way we could call printer.finish() before exiting and get a fully functional HTML report. We no longer would need to write the details to the stdout. Or if we still want to see the report in stdout as well, we could return the assert/error toplist from the post_process function.